internal/gather/servicediscovery.go

```diff
@@ -49,19 +48,11 @@ func gatherCoreDNSPodLogs(info *Info) {
 }
 
 func gatherServiceExports(info *Info, namespace string) {
-	ResourcesToYAMLFile(info, schema.GroupVersionResource{
-		Group:    mcsv1a1.GroupName,
-		Version:  mcsv1a1.GroupVersion.Version,
-		Resource: "serviceexports",
-	}, namespace, metav1.ListOptions{})
+	ResourcesToYAMLFile(info, mcsv1a1.SchemeGroupVersion.WithResource("serviceexports"), namespace, metav1.ListOptions{})
```

Comment:
`mcsv1a1.SchemeGroupVersion` is actually deprecated - I'm guessing newer versions of the code generator don't generate it but the authors kept it for backwards compatibility. I think we should leave this as is - also in other places.

Reply:
They still do, the comment was added following https://github.com/kubernetes/kubernetes/pull/66251#discussion_r203715354 but I don’t know why `SchemeGroupVersion` is supposed to be deprecated (which puts a damper on this whole PR really).

Reply:
yeah and someone asked that over a year ago but wasn't answered. We still need to use `SchemeGroupVersion` for the ones where the code has not been re-generated using a newer generator that deprecates it. So I think this PR is still valid, ie we might as well use `WithResource` where we can w/o being warned about deprecation.